### PR TITLE
Use coverageFile specified by -coverprofile, if given

### PR DESCRIPTION
**Pull request by dhaavi**
Fixes an issue related to https://github.com/joefitzgerald/go-plus/issues/832:

If a -coverprofile is specified in "Additional test args", then use the given filepath.
Possible improvement: Add a random directory between given dir and file. A static coverage file _may_ cause issues, such as race conditions.

I am not an experienced JS developer, so treat this as a Proof of Concept ;)

### CODE REVIEW

File: lib/test/tester.js [hunk 1]
```diff
@@ -259,26 +259,31 @@ class Tester {
   }
 
   createCoverageFile() {
-    this.removeTempDir()
-    if (!this.tempDir) {
-      this.tempDir = fs.realpathSync(temp.mkdirSync())
+    const additionalTestArgs = atom.config.get('go-plus.config.additionalTestArgs')
+    const res = additionalTestArgs.match(/-coverprofile=([A-Za-z0-9/\._-]+)/);
```
Comment (lloiser) [inline lib/test/tester.js]: This regex won't match windows paths because they use `\` instead of `/`.

**Comment by lloiser** [comment]
Hey @dhaavi, thx for the PR!
Can you please rebase it on master?
Regarding the random dir: IMO that's the user's responsibility because she/he provides this argument and should know what she/he does.

File: lib/test/tester.js (hunk 1 above)
Comment (dhaavi) [reply lib/test/tester.js]: Changed to `[^ ]+`.

Comment (dhaavi) [reply lib/test/tester.js]: Thas was a bad idea. It's now `[^\s]`.

**Comment by dhaavi** [comment]
Rebased on master and fixed+tested the new regex.

Regarding the random dir: Yes, you are right - it was just an idea I had.

**Comment by dhaavi** [comment]
Is there a timeline for when this PR will be merged?

**Comment by zmb3** [comment]
Sorry, looks like there’s just some linter errors and I’m happy to merge when those are fixed.

**Comment by dhaavi** [comment]
Thanks @joefitzgerald for fixing this for me, just wanted to finally finish this ;)

**Pull request closed without merging**